paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -655,175 +645,210 @@ abstract class RepositoryLoader<T : LoadableResourceRepository>(
-        )
+        val styleItem = StyleItemResourceValueImpl(namespace, itemName, text, sourceFile.repository.libraryName)
         styleItem.namespaceResolver = itemNamespaceResolver
-        resourceValue.addItem(styleItem)
+        styleItems += styleItem
       }
     }
-
     val visibility = getVisibility(STYLE, name)
-    return BasicResourceItem(STYLE, name, visibility, configuration, resourceValue)
+    val item = BasicStyleResourceItem(name, sourceFile, visibility, parentStyle, styleItems)
+    item.namespaceResolver = namespaceResolver
+    return item
   }
 
   @Throws(IOException::class, XmlPullParserException::class)
   private fun createStyleableItem(
     name: String,
-    file: PathString,
-    configuration: RepositoryConfiguration
-  ): BasicResourceItem {
+    sourceFile: ResourceSourceFile
+  ): BasicStyleableResourceItem {
     val namespaceResolver = parser.namespaceResolver
-    val resourceValue = StyleableResourceValueImpl(namespace, name, null, null)
+    val attrs = mutableListOf<AttrResourceValue>()
     forSubTags(TAG_ATTR) {
       val attrName = parser.getAttributeValue(null, ATTR_NAME)
       if (attrName != null) {
         try {
-          val item = createAttrItem(attrName, file, configuration)
-          val value = item.resourceValue as AttrResourceValue
-          resourceValue.addValue(value)
-
-          if (item.namespace == namespace && (namespace != ResourceNamespace.RES_AUTO || value.formats.isNotEmpty())) {
-            addValueResourceItem(item)
+          val attr = createAttrItem(attrName, sourceFile)
+          // Mimic behavior of AAPT2 and put an attr reference inside a styleable resource.
+          attrs += (if (attr.formats.isEmpty()) attr else attr.createReference())
+
+          // Don't create top-level attr resources in a foreign namespace, or for attr references in the res-auto namespace.
+          // The second condition is determined by the fact that the attr in the res-auto namespace may have an explicit definition
+          // outside of this resource repository.
+          if (attr.namespace == namespace && (namespace != ResourceNamespace.RES_AUTO || attr.formats.isNotEmpty())) {
+            addAttr(attr, attrCandidates)
           }
         } catch (e: XmlSyntaxException) {
-          handleParsingError(file, e)
+          LOG.severe(e.toString())
         }
       }
     }
-
     // AAPT2 treats all styleable resources as public.
     // See https://android.googlesource.com/platform/frameworks/base/+/master/tools/aapt2/ResourceParser.cpp#1539
-    resourceValue.namespaceResolver = namespaceResolver
-    return BasicResourceItem(
-      STYLEABLE,
-      name,
-      ResourceVisibility.PUBLIC,
-      configuration,
-      resourceValue
-    )
+    val item = BasicStyleableResourceItem(name, sourceFile, ResourceVisibility.PUBLIC, attrs)
+    item.namespaceResolver = namespaceResolver
+    return item
+  }
+
+  private fun addAttr(
```

Comment:
looks like this was static and can be moved into the `companion object` right after `isXmlFile(String)` and before `resolveAttrReferences`?